packages/docs/src/components/ColorSwatchList.tsx

```diff
@@ -0,0 +1,59 @@
+import React, { useRef, useLayoutEffect, useState } from 'react';
+
+interface ColorSwatchListProps {
+  colorNames: string[];
+}
+
+// converts an rgb string 'rgb(15,24,128)' to a hex value '#0819A9'
+export const rgbToHex = (r: number, g: number, b: number) => {
+  const hex = ((1 << 24) + (r << 16) + (g << 8) + b).toString(16).slice(1);
+  return `#${hex}`;
+};
+
+const ColorSwatchList = ({ colorNames }: ColorSwatchListProps) => {
+  const refList = useRef([]);
+  const initialColors = colorNames.map((color) => ({ name: color, hex: '' }));
+  const [colorList, setColorList] = useState(initialColors);
+
+  useLayoutEffect(() => {
+    // after swatch has been rendered once, pull rgb color from element styles
+    const updatedColorList = colorList.map((colorItem, index) => {
+      const styles = getComputedStyle(refList.current[index]);
+      const rgbColor = styles?.backgroundColor;
+      if (!rgbColor) return colorItem;
+
+      const rgb = rgbColor.match(/([0-9]+)/g);
+
+      if (rgb.length > 3 && parseInt(rgb[3]) === 0) {
+        // Color is transparent
+        return colorItem;
+      }
+
+      return {
+        ...colorItem,
+        hex: rgbToHex(parseInt(rgb[0]), parseInt(rgb[1]), parseInt(rgb[2])),
+      };
+    });
+
+    setColorList(updatedColorList);
+  }, []);
+
+  return (
+    <div className="c-swatch-list ds-u-border--1 ds-u-padding--2">
+      {colorList.map(({ name, hex }, index) => (
+        <article className="ds-u-margin-bottom--1 c-swatch" key={`${name}-${index}`}>
+          <div
+            className={`c-swatch__preview ds-u-radius--circle ds-u-fill--${name}`}
+            ref={(el) => (refList.current[index] = el)}
+          ></div>
```

Comment:
In the old version, there was a little visual gap between the swatch and the variable name. The new one could probably use some room to breathe:

<img width="87" alt="Screen Shot 2022-05-04 at 4 23 27 PM" src="https://user-images.githubusercontent.com/7595652/166841052-29d63bfb-14cb-4da7-8b0e-30d2a03dd4a8.png">